WordPress/gutenberg
# Lodash: Refactor away from `_.findKey()`

## What?
Lodash's `findKey()` is used only a few times in the entire codebase. This PR aims to remove that usage.

## Why?
Lodash is known to unnecessarily inflate the bundle size of packages, and in most cases, it can be replaced with native language functionality. See these for more information and rationale:

* https://github.com/WordPress/gutenberg/issues/16938#issuecomment-602837246
* https://github.com/WordPress/gutenberg/issues/17025
* https://github.com/WordPress/gutenberg/issues/39495 

## How?
Removing `_.findKey()` is straightforward in favor of a custom implementation that loops through the values and returns the key of the first match.

Essentially, the usages of `findKey()` here were a few, but they were all doing the same thing, so this was a good opportunity to extract and reuse. This seems to have been @ellatrix's intention anyway as mentioned in https://github.com/WordPress/gutenberg/pull/39838#discussion_r837169084. So, we're doing the following here:

* Abstracting the block attribute selection position retrieval to a separate function.
* Adding tests for that function, in order to ensure we're safely removing Lodash later.
* Refactoring the function from `_.findKey()`, which was the sole usage of `findKey()`.
* Deprecating `_.findKey()` in our ESLint config so it won't be used again.

## Testing Instructions

Verify tests pass: `npm run test-unit packages/block-editor/src/utils/test/selection.js`
